### PR TITLE
add an option to inject examples

### PR DESCRIPTION
**Pull request by keertip**
This adds a flag `--include-examples` which for now looks for `@example` tag in the comments and looks for the file associated with the sample code.

#1105

### CODE REVIEW

File: lib/src/model.dart [hunk 1]
```diff
@@ -1605,6 +1611,27 @@ abstract class ModelElement implements Comparable, Nameable, Documentable {
...
+  String _injectExamples(String rawdocs) {
+    if (rawdocs.contains('@example')) {
+      RegExp exp = new RegExp(r"{@example .+}");
+      Iterable<Match> matches = exp.allMatches(rawdocs);
+      var dirPath = this.package.packageMeta.dir.path;
+      for (var match in matches) {
+        var strings = match.group(0).split(' ');
+        var file = new File(p.join(dirPath, 'examples', strings[1]));
+        if (file.existsSync()) {
+          // TODO(keertip):inject example
```
Comment (sethladd) [inline lib/src/model.dart]: did we mean to fill this out?

File: testing/test_package/lib/example.dart [hunk 2]
```diff
@@ -162,6 +162,7 @@ class ConstantCat implements Cat {
 }
 
 /// implements [Cat], [E]
+/// {@example core/pipes/ts/slice_pipe/slice_pipe_example.ts region='SlicePipe_list'}
```
Comment (sethladd) [inline testing/test_package/lib/example.dart]: What is this path relative to? We might want to document that fact in the main source for this feature?

File: bin/dartdoc.dart [hunk 3]
```diff
@@ -195,6 +180,11 @@ ArgParser _createArgsParser() {
       allowMultiple: true, help: 'library names to generate docs for');
   parser.addOption('include-external',
       allowMultiple: true, help: 'additional (external) libraries to include');
+  parser.addFlag('include-examples',
+      defaultsTo: false,
+      negatable: false,
+      help:
+          "look for '@example' tags and inject the examples into the comment");
```
Comment (sethladd) [inline bin/dartdoc.dart]: Add a note for how to find an example? Or, show an example here? Also, how are the relative paths found?

File: lib/src/model.dart (hunk 1 above)
Comment (keertip) [inline lib/src/model.dart]: Yes. Currently there are no examples in Angular2 dart, getting this in will give them a list of the missing examples.

File: testing/test_package/lib/example.dart (hunk 2 above)
Comment (keertip) [inline testing/test_package/lib/example.dart]: relative to the package. Yes, will document.

File: bin/dartdoc.dart (hunk 3 above)
Comment (keertip) [inline bin/dartdoc.dart]: done.

**Comment by devoncarew** [comment]
Two drive by comments:

- using both `--include-examples` and `@example` seems over-specified; what about just keying off a well-formed `@example` annotation?
- the other directive we recognize is `<nodoc>`; that's not consistent with the `@` annotation above; perhaps use the same style of directive?

**Comment by keertip** [comment]
Thanks Devon for the comments!

Will remove the flag. For the `<nodoc>`, we can add support for a `@nodoc` format to make it consistent.

**Pull request merged**